packages/manager/modules/sign-up/src/form/form.controller.js

```diff
@@ -102,6 +102,31 @@ export default class SignUpFormCtrl {
           }
         });
 
+        this.rules.legalform.in = this.rules.legalform.in
+          .reduce(
+            (prev, cur) => {
+              switch (cur.value) {
+                case 'administration':
+                  prev.splice(2, 1, cur);
+                  break;
+                case 'association':
+                  prev.splice(3, 1, cur);
+                  break;
+                case 'corporation':
+                  prev.splice(0, 1, cur);
+                  break;
+                case 'individual':
+                  prev.splice(1, 1, cur);
+                  break;
+                default:
+                  prev.push(cur);
+              }
+              return prev;
+            },
+            [null, null, null, null],
+          )
+          .filter((el) => el);
```

Comment:
If I am correct, the output after reduce will be
["corporation", "individual", "administration", "association"]
Is this what you are trying to get?

Reply:
["corporation", "individual", "administration", "association", "other"] for Fr, but for US we dant have "administration" for exemple and i'm trying to make a résilient order if API add other.

Reply:
i think it should be simpler, maybe take a look here : https://stackoverflow.com/questions/16221197/sort-javascript-array-in-a-pre-defined-order

Reply:
What I see from the code is that you are trying to sort the values in a particular order. Is that all required here?

Reply:
yes we want order them : ["corporation", "individual", "administration", "association", "other"]
but i want to make it resilient and if API change, this is working. For exemple remove Organisation and add "gaming" for exemple. 

It's useless maybe, so i can change method for the @frenautvh ref, the first thread awnser crawl the array much time (each indexOf + ref array) where i crawl array 2 time. So i'm looking for second one (but it need to be a predictible array)

Reply:
@Heiiji i think it doesn't matter if the first thread answer is slower that your solution, the array is so small that there is no need to seek performance, you shoud favor code readability in this case